src/Methods/Update.php

```diff
@@ -220,4 +224,26 @@ private function getRequestBody(?RequestInterface $request)
 
         return json_decode($rawBody, true);
     }
+
+    /** Dispatch Update Event. */
+    private function dispatchUpdateEvent(UpdateObject $update): void
+    {
+        if (! property_exists($this, 'eventEmitter') || ! $this->eventEmitter instanceof EmitterInterface) {
+            return;
+        }
+
+        $eventEmitter = $this->eventEmitter;
+
+        $event = new UpdateEvent($this, $update);
+
+        $eventEmitter->emit($event->cloneWithCustomName(UpdateEvent::NAME));
+        $updateType = $update->objectType();
+        if (is_string($updateType)) {
+            $eventEmitter->emit($event->cloneWithCustomName($updateType));
+
+            if (null !== $update->getMessage()->objectType()) {
+                $eventEmitter->emit($event->cloneWithCustomName($updateType . '.' . $update->getMessage()->objectType()));
+            }
+        }
```

Comment:
side by side comparison with SDK v4 (if it will help you to review):
<img width="1982" alt="image" src="https://user-images.githubusercontent.com/5278175/183777369-961611f8-f951-4966-b0a4-91cc52fc09cb.png">